code/modules/antagonists/ninja/ninjaDrainAct.dm

```diff
@@ -296,8 +295,23 @@
 	modelselected["Medical"] = "/obj/item/robot_model/ninja/ninja_medical"
 	modelselected["Saboteur"] = "/obj/item/robot_model/ninja_saboteur"
 	//SKYRAT EDIT: ADDITION END
+
 	if(!do_after(ninja, 6 SECONDS, target = src))
 		return
+	//ZUBBER EDIT START - Puts a cap on ninja's borg hacking
+	//Set to negative numbers for infinite uses
+	var/datum/antagonist/ninja/borg_hacks_left = -1
```

Comment:
wait. you're making an uninitialized datum reference, and assigning it's object to be -1? that doesn't seem right. you shouldn't need to typecast anything here.